libsolidity/interface/StandardCompiler.cpp

```diff
@@ -131,6 +131,62 @@ StringMap createSourceList(Json::Value const& _input)
 	return sources;
 }
 
+bool isTargetRequired(Json::Value const& _targets, string const& _target)
+{
+	for (auto const& target: _targets)
+		/// @TODO support sub-macthing, e.g "evm" matches "evm.assembly"
+		if (target == "*" || target == _target)
+			return true;
+	return false;
+}
+
+///
+/// @_targets is a JSON object containining a two-level hashmap, where the first level is the filename,
+/// the second level is the contract name and the value is an array of target names to be requested for that contract.
+/// @_file is the current file
+/// @_contract is the current contract
+/// @_target is the current target name
+///
+/// Returns true if the _targets has a match for the requested target in the specific file / contract.
+///
+/// In @_targets the use of '*' as a wildcard is permitted.
+///
+/// @TODO optimise this. Perhaps flatten the structure upfront.
+///
+bool isTargetRequired(Json::Value const& _targets, string const& _file, string const& _contract, string const& _target)
```

Comment:
Better naming: `isArtifactRequired(_outputSelection, _file, _contract, _artifactName)`

Reply:
Fine with that, but would perhaps use `isArtifactRequested`